diff --git a/lib/DocApi.php b/lib/DocApi.php
index 1bec149..94f8c69 100644
--- a/lib/DocApi.php
+++ b/lib/DocApi.php
@@ -401,7 +401,9 @@ class DocApi
     public function createDoc($doc)
     {
         list($response) = $this->createDocWithHttpInfo($doc);
-        return $response;
+        // Patch to preserve old behavior of returning binary data.
+        // return $response;
+        return file_get_contents($response->getPathname());
     }
 
     /**
@@ -1445,7 +1447,9 @@ class DocApi
     public function getAsyncDoc($id)
     {
         list($response) = $this->getAsyncDocWithHttpInfo($id);
-        return $response;
+        // Patch to preserve old behavior of returning binary data.
+        // return $response;
+        return file_get_contents($response->getPathname());
     }
 
     /**
